server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/filter/JMAPFilteringTest.java

```diff
@@ -0,0 +1,3393 @@
+
+        void mailDirectiveShouldBeSetWhenMultipleHeaders(JMAPFilteringTestSystem testSystem) throws Exception;
+    }
+
+    interface WithNotExactlyEqualsCommandRule {
+        void mailDirectiveShouldBeSetWhenDoNotExactlyEqualsRuleValue(JMAPFilteringTestSystem testSystem) throws Exception;
+
+        void mailDirectiveShouldBeSetWhenAllDoNotExactlyEqualsRuleValue(JMAPFilteringTestSystem testSystem) throws Exception;
+
+        void mailDirectiveShouldNotBeSetWhenExactlyEqualsRuleValue(JMAPFilteringTestSystem testSystem) throws Exception;
+
+        void mailDirectiveShouldBeSetWhenUnscrambledContentDoesntExactlyEqualsRuleValue(JMAPFilteringTestSystem testSystem) throws Exception;
+
+        void mailDirectiveShouldBeSetWhenFoldedContentDoesntExactlyEqualsRuleValue(JMAPFilteringTestSystem testSystem) throws Exception;
+
+        void mailDirectiveShouldBeSetWhenMultipleHeaders(JMAPFilteringTestSystem testSystem) throws Exception;
+    }
+
+    interface WithRecipientFiltering {
+        void mailDirectiveShouldIgnoreBccHeaders(JMAPFilteringTestSystem testSystem) throws Exception;
+    }
+
+    private static final String DELIVERY_PATH_PREFIX = "DeliveryPath_";
+
+    private static final Rule.Id RULE_ID_1 = Rule.Id.of("1");
+    private static final String RULE_NAME_1 = "rule 1";
+
+    private static final String SENDER_1 = "sender1@james.org";
+    private static final String SENDER_1_USERNAME = "sender1";
+
+    private static final String SENDER_2 = "sender2@james.org";
+
+    private static final String RECIPIENT_1 = "recipient1@james.org";
+    static final String RECIPIENT_1_USERNAME = "recipient1";
+    static final String RECIPIENT_1_MAILBOX_1 = "recipient1_maibox1";
+
+    private static final String RECIPIENT_2 = "recipient2@james.org";
+
+    static final String FRED_MARTIN_USERNAME = "fred.martin";
+    static final String FRED_MARTIN_INBOX = "fred.martin.inbox";
+    private static final String FRED_MARTIN = "fred.martin@linagora.com";
+
+    @Nested
+    class FromFiltering {
+
+        @Nested
+        class FromFilteringWithContainsCommandRule implements WithContainsCommandRule {
+
+            @Test
+            @Override
+            public void mailDirectiveShouldBeSetWhenContainsRuleValue(JMAPFilteringTestSystem testSystem) throws Exception {
+                testSystem.getFilteringManagement().defineRulesForUser(User.fromUsername(RECIPIENT_1_USERNAME),
+                        ImmutableList.of(
+                            Rule.builder()
+                                .id(RULE_ID_1)
+                                .name(RULE_NAME_1)
+                                .condition(Rule.Condition.of(Rule.Condition.Field.FROM, Rule.Condition.Comparator.CONTAINS, SENDER_1_USERNAME))
+                                .action(Rule.Action.of(Rule.Action.AppendInMailboxes.withMailboxIds(ImmutableList.of(testSystem.getRecipient1MailboxId().serialize()))))
```

Comment:
We can add a `AppendInMailboxes withMailboxIds(String... )` method to make our tests more readable